### PR TITLE
stream: use SO_OOBINLINE on OS X

### PR DESCRIPTION
**Pull request by indutny**
In the collaboration with Ben Noordhuis <info@bnoordhuis.nl> and
Saúl Ibarra Corretgé <saghul@gmail.com>.

@saghul please process this urgently

### CODE REVIEW

File: test/test-tcp-oob.c [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  int r;
+  uv_os_fd_t fd;
+
+  ASSERT(0 == status);
+  ASSERT(0 == uv_accept(handle, (uv_stream_t*) &peer_handle));
+  ASSERT(0 == uv_read_start((uv_stream_t*) &peer_handle, alloc_cb, read_cb));
+
+  /* Send some OOB data */
+  ASSERT(0 == uv_fileno((uv_handle_t*) &client_handle, &fd));
+
```
Comment (saghul) [inline test/test-tcp-oob.c]: The problem triggered with more than one message, right? If so, can you please add a comment clarifying that here?

**Comment by bnoordhuis** [comment]
LGTM.  Great test case!

File: test/test-tcp-oob.c (hunk 1 above)
Comment (indutny) [inline test/test-tcp-oob.c]: Done.

File: test/test-tcp-oob.c [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  ASSERT(0 == uv_stream_set_blocking((uv_stream_t*) &client_handle, 1));
+
+  do {
```
Comment (saghul) [inline test/test-tcp-oob.c]: The problem triggered with more than one message, right? If so, can you please add a comment clarifying that here?

**Comment by indutny** [comment]
@bnoordhuis ahaha! :) With your help, as cited.

**Comment by saghul** [comment]
LGTM. If you squash and land I can make a v1.x release. Also we'll need a backport for 0.10.

**Comment by indutny** [comment]
@saghul thanks, landed in e19089f . Please release (and maybe backport it ;) ? )

**Pull request closed without merging**